app/Community/Controllers/MessageThreadController.php

```diff
@@ -36,6 +40,29 @@ public function index(Request $request): InertiaResponse|RedirectResponse
         return Inertia::render('messages', $actionResult['props']);
     }
 
+    public function show(Request $request, MessageThread $messageThread): InertiaResponse|RedirectResponse
+    {
+        $this->authorize('view', $messageThread);
```

Comment:
Trying to access a deleted message thread generates an "Unauthorized" error instead of a "Not Found" error.

Reply:
Addressed in latest.